### PR TITLE
OOP-SCHOOL-library

### PR DESCRIPTION
**Pull request by ambrose-kibet**
## In this milestone we:
- [ ] Create class Person
- [ ] Create class Student
- [ ] Create class Teacher

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(name, age, parent_permission, classroom)
+    super(name, age, parent_permission)
+
+    @classroom = classroom
+  end
+
```
Comment (DammyShittu) [inline student.rb]: - [x] Kindly ensure you are mindful of the order of the parameters. In the initialize parameters, `age` should come before `name`, and `classroom` should come before `name`, and `parent_permission`.

- [x] Remember that name and parent_permission are optional, so we should indicate that.

- [ ] In the super method, age, should come before name based on how the Person class is defined.

- [x] In the super method list of parameters, the `parent_permission` should also be defined correctly since it uses the colon notation.

  In summary this should be your code:

  ```
  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
    @classroom = classroom
  end
  ```

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Student < Person
```
Comment (DammyShittu) [inline student.rb:2]: - [x] Kindly import the person class to this file for this inheritance to work.

  ```
  require_relative './person'
  ```

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(name, age, specialization, parent_permission: true)
+    super(name, age, parent_permission)
+
+    @specialization = specialization
+  end
```
Comment (DammyShittu) [inline teacher.rb]: - [x] Kindly ensure you are mindful of the order of the parameters. In the initialize parameters, `age` should come before `name`, and `specialization` should come before `name`, and `parent_permission`.

- [x] Remember that `name` and `parent_permission` are optional, so we should indicate that.

- [x] In the super method, `age` should come before `name` based on how the Person class is defined.

- [x] In the super method list of parameters, the `parent_permission` should also be defined correctly since it uses the colon notation.

  In summary this should be your code:

  ```
  def initialize(age, specialization, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
    @specialization = specialization
  end
  ```

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class Teacher < Person
```
Comment (DammyShittu) [inline teacher.rb:2]: - [x] Kindly import the person class to this file for the inheritance to work.

  ```
  require_relative './person'
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @ambrose-kibet,

## Highlights
✔️ Neat person class

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb (hunk 1 above)
Comment (KDlamini) [reply student.rb]: Well resolved ✔️

File: student.rb (hunk 2 above)
Comment (KDlamini) [reply student.rb:2]: Well resolved ✔️

File: teacher.rb (hunk 3 above)
Comment (KDlamini) [reply teacher.rb]: Well resolved ✔️

File: teacher.rb (hunk 4 above)
Comment (KDlamini) [reply teacher.rb:2]: Well resolved ✔️

**Review by KDlamini — APPROVED** [review]
### Hi @ambrose-kibet 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ Professional pull request
✔️ There is an individual `.rb` file for each class (`Person`, `Student`, and `Teacher` ).
✔️ Class inheritance is well-implemented

# Status: Approved ✔️ 

Great-job-smiley

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
